patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -572,7 +572,7 @@
 +					ManaEffect(healMana);
  			}
  
-+			if (ItemLoader.ConsumeItem(item, this))
++			if (item.consumable && ItemLoader.ConsumeItem(item, this))
```

Comment:
Instead of changing both of these locations, 
Add the ' not item.consumable' to ItemLoader.ConsumeItem method itself

Ensures the fix is applied in all locations.

Reply:
`ItemLoader.ConsumeItem` has 34 references. For quick mana/health it makes sense to check `item.consumable`, because the intent is to consume a potion or edible. 

For many others (placeables, gold keys, paint, wire, etc etc) it would be breaking change to check `item